seacatauth/authn/handler.py

```diff
@@ -256,3 +255,33 @@ async def smslogin(self, request):
 
 		body = {"result": "OK" if success is True else "FAILED"}
 		return aiohttp.web.Response(body=login_session.encrypt(body))
+
+
+	async def webauthn_login(self, request):
+		# Decode JSON request
+		lsid = request.match_info["lsid"]
+		login_session = self.AuthenticationService.LoginSessions.get(lsid)
+		if login_session is None:
+			L.error("Login session not found.", struct_data={"lsid": lsid})
+			raise aiohttp.web.HTTPUnauthorized()
+
+		json_body = login_session.decrypt(await request.read())
+
+		# descriptor_id = json_body.get("descriptor_id")
+		factor_type = json_body.get("factor_type")
+		if factor_type != "webauthn":
+			body = {"result": "FAILED", "message": "Unsupported factor type."}
+			return aiohttp.web.Response(body=login_session.encrypt(body))
+
+		# Webauthn challenge timeout should be the same as the current login session timeout
+		timeout = (login_session.ExpiresAt - datetime.datetime.now()).total_seconds() * 1000
```

Comment:
`datetime.datetime.now` ... is this UTC?

Reply:
good catch. fixed.